lib/mayaUsd/render/vp2RenderDelegate/render_delegate.h

```diff
@@ -173,6 +175,8 @@ class HdVP2RenderDelegate final : public HdRenderDelegate
     static HdResourceRegistrySharedPtr
         _resourceRegistry; //!< Shared and unused by VP2 resource registry
 
+    std::unordered_set<HdSprim*> _materialSprims;
```

Comment:
Do we ever get sprims which are _not_ HdVP2Material? Having to do the dynamic cast is kind of a bummer. We only DO use this on clearing the scene so maybe the performance isn't a huge deal.

Is there no way for Usd to give us an iterator over all sprims?

Reply:
I can't find any API in SceneDelegate class that could return all created sprims :-(